spec/listing/listing.spec.ts

```diff
@@ -1,53 +1,214 @@
 import * as ganache from 'ganache-cli'
 import Web3 from 'web3'
 import { Contract } from 'web3/types'
-import Voting from '../../src/contracts/voting'
-import Parameterizer from '../../src/contracts/parameterizer'
 import Listing from '../../src/contracts/listing'
+import { LISTING_ABI, ONE_ETHER, ONE_GWEI } from  '../../src/constants'
+import { Return } from '../../src/@types'
+import {
+  deploy,
+  readBytecode,
+  call,
+  transact,
+} from '../../src/helpers'
 
 const provider:any = ganache.provider(),
-  w3 = new Web3(provider)
+  w3 = new Web3(provider, undefined, {defaultBlock: 'latest',
+    transactionConfirmationBlocks: 1, transactionBlockTimeout: 5}),
+    toBN = w3.utils.toBN
 
-let accounts:string[],
-  voting:Voting,
-  parameterizer:Parameterizer,
-  listing:Listing
+let listing:Listing,
+  accounts:string[],
+  deployed:Contract
 
 describe('Listing', () => {
-  beforeAll(() => {
-
-  })
-
-  afterAll(() => {
-
-  })
-
-  beforeEach(async () => {
+  beforeAll(async () => {
     accounts = await w3.eth.getAccounts()
 
-    // erc20 = new Erc20(accounts[0])
+    // deploy it...
+    const bin:string = readBytecode('voting')
 
-    // voting = new Voting(accounts[0])
+    deployed = await deploy(w3,
+                            accounts[0],
+                            LISTING_ABI,
+                            bin,
+                            [accounts[0], accounts[0], accounts[0], accounts[0], accounts[0]])
```

Comment:
hmm. so let's not do this. I think we'll likely get into a situation really soon where we make real calls (if only a few) and this contract should be setup with that expectation from the start.

since we know it expects addresses in the order: `(market, voting, p11r, reserve, datatrust)` let's go ahead and provide them thus.

now, I don't think you really need to deploy those just yet, you can put fake addresses on them
`const marketAddress = whatever` - but at least later when we update that assignment to an actually deployed market token the constructor will still work

Reply:
done.